include/boost/math/distributions/von_mises.hpp

```diff
@@ -0,0 +1,842 @@
+namespace boost{ namespace math{
+
+template <class RealType = double, class Policy = policies::policy<> >
+class von_mises_distribution
+{
+public:
+  typedef RealType value_type;
+  typedef Policy policy_type;
+
+  von_mises_distribution(RealType l_mean = 0, RealType concentration = 1)
+    : m_mean(l_mean), m_concentration(concentration)
+  { // Default is a 'standard' von Mises distribution vM01.
+    static const char* function = "boost::math::von_mises_distribution<%1%>::von_mises_distribution";
+
+    RealType result;
+    detail::check_positive_x(function, concentration, &result, Policy());
+    detail::check_angle(function, l_mean, &result, Policy());
+  }
+
+  RealType mean() const
+  { // alias for location.
+    return m_mean;
+  }
+
+  RealType concentration() const
+  { // alias for scale.
+    return m_concentration;
+  }
+
+  // Synonyms, provided to allow generic use of find_location and find_scale.
+  RealType location() const
+  { // location.
+    return m_mean;
+  }
+  RealType scale() const
+  { // scale.
+    return m_concentration;
+  }
+
+private:
+  //
+  // Data members:
+  //
+  RealType m_mean;  // distribution mean or location.
+  RealType m_concentration;    // distribution standard deviation or scale.
+}; // class von_mises_distribution
+
+typedef von_mises_distribution<double> von_mises;
+
+#ifdef BOOST_MSVC
+#pragma warning(push)
+#pragma warning(disable:4127)
+#endif
+
+template <class RealType, class Policy>
+inline const std::pair<RealType, RealType> range(const von_mises_distribution<RealType, Policy>& /*dist*/)
+{ // Range of permissible values for random variable x.
+  if (std::numeric_limits<RealType>::has_infinity)
+  { 
+    return std::pair<RealType, RealType>(-std::numeric_limits<RealType>::infinity(),
+                                          +std::numeric_limits<RealType>::infinity()); // - to + infinity.
+  }
+  else
+  { // Can only use max_value.
+    using boost::math::tools::max_value;
+    return std::pair<RealType, RealType>(-max_value<RealType>(),
+                                         +max_value<RealType>()); // - to + max value.
+  }
+}
+
+template <class RealType, class Policy>
+inline const std::pair<RealType, RealType> support(const von_mises_distribution<RealType, Policy>& dist)
+{ // This is range values for random variable x where cdf rises from 0 to 1, and outside it, the pdf is zero.
+  return std::pair<RealType, RealType>(dist.mean() - constants::pi<RealType>(),
+                                       dist.mean() + constants::pi<RealType>()); //  [µ-π, µ+π)
+}
+
+#ifdef BOOST_MSVC
+#pragma warning(pop)
+#endif
+
+namespace detail {
+// float version of pdf_impl
+template <class RealType, class Policy>
+inline RealType pdf_impl(const von_mises_distribution<RealType, Policy>& dist, RealType x,
+                         const boost::integral_constant<int, 24>&)
+{
+  RealType mean = dist.mean();
+  RealType conc = dist.concentration();
+
+  BOOST_MATH_STD_USING
+  if(conc < 7.75)
```

Comment:
I guess `conc < 7.75` is a cutoff of an asymptotic expansion? Could you add a comment if so?